### PR TITLE
error out when --reuse-key conflicts with other flags

### LINKED ISSUE #7711: --reuse-key ambiguity
If you run
```
sudo certbot -d example.org --standalone certonly
```
Certbot will obtain a certificate using a 2048 bit RSA key.

If you then run
```
sudo certbot -d example.org --standalone certonly --force-renewal --rsa-key-size 4096 --reuse-key
```
the current 2048 bit RSA key is reused in the new certificate and the resulting renewal configuration file contains the following renewal parameters
```
reuse_key = True
rsa_key_size = 4096
```
I don't think this is the correct behavior. We had a brief [conversation about this in Mattermost](https://opensource.eff.org/eff-open-source/pl/4qo67zuj7brcfbjaxsecqy3our), where most people thought erroring out was the correct behavior.

This probably seems right, however, I then think we need a way to specify that a new key should be created and that key should be used for subsequent renewals. Peng suggested maybe using a separate flag for this?

I think we should fix this, however, I think this is technically a breaking change and should be done a little cautiously.

### PR DESCRIPTION
**Pull request by alexzorin**
This is the second half of the fix for #7711. 

Fixes #7711.

Because it is a breaking change, it should be merged right before the release of 2.0.

- [x] Confirm that it errors in [all relevant scenarios](https://docs.google.com/spreadsheets/d/1B1XbsCsgvBOP0TlYhFFXNaQoJTVe_7UEfI_WphaeMp8/edit#gid=0) for `certonly`/`run` and `renew`
- [x] Write unit and integration tests
- [ ] Test in conjunction with the transition to `ecdsa` by default
- [x] Write a CHANGELOG entry for 2.0

### CODE REVIEW

**Comment by bmw** [comment]
This looks great to me Alex. Nice work as usual.

I think the only things we need to do are ensure a changelog entry is added about this for 2.0 like you put in your checklist and to merge this into a 2.0 branch instead of `master`.

**Review by bmw — APPROVED** [review]

**Comment by alexzorin** [comment]
Thanks. As for testing this in conjunction with #9353, I think we can do that directly from the `2.0.x` branch.

File: certbot/certbot/_internal/renewal.py [hunk 1]
```diff
@@ -326,12 +326,57 @@ def _avoid_invalidating_lineage(config: configuration.NamespaceConfig,
...
+    kt = config.key_type.lower()
+
+    # The remaining cases where conflicts are present:
+    # - --key-type is set on the CLI and doesn't match the stored private key
+    # - It's an RSA key and --rsa-key-size is set and doesn't match
+    # - It's an ECDSA key and --eliptic-curve is set and doesn't match
+    potential_conflicts = [
+        ("--key-type",
+         lambda: kt != lineage.private_key_type.lower()),
+        ("--rsa-key-type",
```
Comment (alexzorin) [inline certbot/certbot/_internal/renewal.py:357]: Blah. This is meant to be `--rsa-key-size`. I'll fix it in #9353.

Comment (bmw) [reply certbot/certbot/_internal/renewal.py:357]: Nice catch!

**Pull request merged**